### ISSUE TITLE
Inefficient LINQ projections

### ISSUE THREAD

**Issue by Vannevelj** [Vannevelj/VSDiagnostics#646]
```csharp
.Select().Take()
.Select().Skip()
.Select().TakeWhile()
.Select().SkipWhile()
.Select().First()
.Select().Single()
.Select().Last()
.Select().FirstOrDefault()
.Select().SingleOrDefault()
.Select().LastOrDefault()
```

Closely related to https://github.com/Vannevelj/VSDiagnostics/issues/645

**Comment by Hosch250** [comment]
Note that when using Entity Framework, these aren't inefficient, as they translate into a SQL script, and can be more efficient in that you can tell it to return fewer fields with the `Select`.

**Comment by Hosch250** [comment]
Update based on the previous comment: if we don't warn on `IQueryable`, but only `IEnumerable`, it should be good.
